src/Appium.Net/Appium/Service/AppiumServiceBuilder.cs

```diff
@@ -450,6 +478,19 @@ private string Args
             {
                 List<string> argList = new List<string>();
                 CheckAppiumJS();
+
+                if (NodeOptions != null)
+                {
+                    foreach (var argPair in NodeOptions)
+                    {
+                        argList.Add(argPair.Key);
+                        if (!string.IsNullOrEmpty(argPair.Value))
+                        {
+                            argList.Add(argPair.Value);
```

Comment:
should there be a check if value has spaces?

Reply:
Fixed